pkg/execution/state/v2/interfaces.go

```diff
@@ -33,7 +33,7 @@ type RunService interface {
 	// Delete deletes state, metadata, and - when pauses are included - associated pauses
 	// for the run from the store.  Nothing referencing the run should exist in the state
 	// store after.
-	Delete(ctx context.Context, id ID) error
+	Delete(ctx context.Context, id ID) (bool, error)
```

Comment:
cc @BrunoScheufler - sharing context as it's a change to the state interfaces we'll need to account for when we upstream this.